lib/connectors/memory.js

```diff
@@ -148,11 +148,13 @@ Memory.prototype.saveToFile = function (result, callback) {
   if (file) {
     if(!self.writeQueue) {
       // Create a queue for writes
+      var cache = self.cache;
+      var ids = self.ids;
       self.writeQueue = async.queue(function (task, cb) {
         // Flush out the models/ids
         var data = JSON.stringify({
-          ids: self.ids,
-          models: self.cache
+          ids: ids,
+          models: cache
```

Comment:
@bajtos 

OK that's my conclusion after lots of investigation; I have always had this problem in memory connector not only in my feature branch, but also in master branch... The problem that I experience is you can `create` a record using memory and saving in a file for the first time you run the app and it creates the file and saves your record(s) in the file. but if you restart the server and use the same file for your DB and add another record with another id, it does not save it in the file anymore.

As you see I see different data inside and outside of `async.queue`
```
      console.log('> cache:', self.cache, 'ids:', self.ids); // It shows the updated data
      self.writeQueue = async.queue(function (task, cb) {
        console.log('> cache:', self.cache, 'ids:', self.ids); // It shows the old data
        var data = JSON.stringify({
          ids: self.ids,
          models: self.cache
        }, null, '  ');

        fs.writeFile(self.settings.file, data, function (err) {
          cb(err);
          task.callback && task.callback(err, task.data);
        });
      }, 1);
```

However, although this does not work for me on my work PC, I used the same juggler version or even this feature branch in juggler in two other computers and it works like a charm. It is worth mentioning we all use  node 4.2.2.

In terms of testing, first of all I'm not sue how I can reproduce the issue, because it happens just when you restart the server, secondly it seems it passes on other computers except my work PC...